python/dgl/graphbolt/impl/sampled_subgraph_impl.py

```diff
@@ -68,6 +68,63 @@ def __post_init__(self):
             ), "Nodes in pairs should be of type torch.Tensor."
 
 
+def _to_reverse_ids(node_pair, reverse_row_node_ids, reverse_column_node_ids):
+    u, v = node_pair
+    if reverse_row_node_ids is not None:
+        u = reverse_row_node_ids[u]
+    if reverse_column_node_ids is not None:
+        v = reverse_column_node_ids[v]
+    return (u, v)
+
+
+def _relabel_homo_edges(lhs_edges, rhs_edges):
+    """Relabel the source nodes and destination nodes independently for two set
+    of edges."""
+    src = torch.cat([lhs_edges[0], rhs_edges[0]])
+    dst = torch.cat([lhs_edges[1], rhs_edges[1]])
+    _, src = torch.unique(src, return_inverse=True)
+    _, dst = torch.unique(dst, return_inverse=True)
+    lhs_edges = (src[: lhs_edges[0].numel()], dst[: lhs_edges[0].numel()])
+    rhs_edges = (src[lhs_edges[0].numel() :], dst[lhs_edges[0].numel() :])
+    return lhs_edges, rhs_edges
+
+
+def _exclude_homo_edges(lhs_edges, rhs_edges):
+    lhs_edges, rhs_edges = _relabel_homo_edges(lhs_edges, rhs_edges)
+    # Compact edges to integers to use torch.isin to exclude edges
+    dst_range = lhs_edges[0].numel() + rhs_edges[0].numel()
+    lhs_val = lhs_edges[0] * dst_range + lhs_edges[1]
+    rhs_val = rhs_edges[0] * dst_range + rhs_edges[1]
+    mask = ~torch.isin(lhs_val, rhs_val)
+    return torch.nonzero(mask, as_tuple=True)[0]
+
+
+def _slice_subgraph(subgraph: SampledSubgraphImpl, index: torch.Tensor):
+    """Slice the subgraph according to the index."""
+
+    def _index_select(obj, index):
+        if obj is None:
+            return None
+        if isinstance(obj, torch.Tensor):
+            return obj[index]
+        if isinstance(obj, tuple):
+            return tuple(_index_select(v, index) for v in obj)
+        # Handle the case when obj is a dictionary
+        assert isinstance(obj, dict)
+        assert isinstance(index, dict)
+        ret = {}
+        for k, v in obj.items():
+            ret[k] = _index_select(v, index[k])
+        return ret
+
+    return SampledSubgraphImpl(
```

Comment:
Is the returned SampledSubgraphImpl compacted or not or depends on the input SampledSubgraphImpl?

Reply:
It depends on the input subgraph. I will add explanation about this.